blocks.go

```diff
@@ -50,6 +50,43 @@ func (srv *notesAPI) InsertBlock(ctx context.Context, req *notesv1.InsertBlockRe
 	return &notesv1.InsertBlockResponse{Block: modelsBlockToProtobufBlock(block)}, nil
 }
 
+func (srv *notesAPI) UpdateBlockIndex(ctx context.Context, req *notesv1.UpdateBlockIndexRequest) (*notesv1.UpdateBlockIndexResponse, error) {
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validators.ValidateUpdateBlockIndexRequest(req)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	block, err := srv.notes.GetBlock(ctx,
+		&models.OneBlockFilter{GroupID: req.GroupId, NoteID: req.NoteId, BlockID: req.BlockId},
+		token.AccountID)
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+	err = srv.notes.DeleteBlock(ctx,
+		&models.OneBlockFilter{GroupID: req.GroupId, NoteID: req.NoteId, BlockID: req.BlockId},
+		token.AccountID)
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+	block, err = srv.notes.InsertBlock(ctx,
```

Comment:
there is now way to update the block without deleting and inserting a new one ?

Reply:
Je l'ai écris dans la description de la PR,
En gros le seul moyen d'inserer le block a une position c'est lors de la création, sinon on peut juste update ses champs.
Donc le seul moyen de changer l'ordre des blocks sans delete & create, c'est d'avoir un field `index` dans les blocks (stocké en db) et lorsqu'on insert un bloque on doit faire toute la logique de bouger les indexs. 
Ex : Si j'insère un new block en 0 il faut que l'ancien block en 0 soit en n+1 etc etc.
Et dans le front pareil faut les trier en fonction de l'index. Ou le faire dans getNote mais sachant qu'on a la logique d'index dans le front ça reviens au même. On aura des références aux index en front.

Donc en faisant ca je m'évite toute la logique back.
Sachant que je l'avais déjà implem a l'ancienne mais diego a tout tej pour mettre les blocks dans une note. Pour que les id des blocks soit pas unique.